gemini/use-cases/applying-llms-to-data/docai-gemini/cf-trigger-event/src/extractor.py

```diff
@@ -0,0 +1,147 @@
+import re
+from typing import Optional
+
+from google.api_core.client_options import ClientOptions
+from google.api_core.exceptions import InternalServerError, RetryError
+from google.cloud import documentai, storage
+from temp_file_uploader import TempFileUploader
+
+
+class DocumentExtractor:
+    def __init__(
+        self,
+        project_id: str,
+        location: str,
+        processor_id: str,
+        processor_version_id: Optional[str] = None,
+    ):
+        self.project_id = project_id
+        self.location = location
+        self.processor_id = processor_id
+        self.processor_version_id = processor_version_id
+        self.client = documentai.DocumentProcessorServiceClient(
+            client_options=ClientOptions(
+                api_endpoint=f"{location}-documentai.googleapis.com"
+            )
+        )
+        self.processor_name = self._get_proccessor_name()
```

Comment:
I'd recommend making this a `@property` https://www.programiz.com/python-programming/property